src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -19,6 +19,8 @@ namespace Microsoft.CodeAnalysis
 {
     internal partial class DocumentState : TextDocumentState
     {
+        private static readonly Func<string, PreservationMode, string> s_fullParseLog = (path, mode) => $"{path} : {mode}";
```

Comment:
s_fullParseLog won't be executed until logging is enabled with verbose mode. so basically it never happens except someone want to explicitly enable it to investigate something  like perf. so allocation is not an issue here.